### PR TITLE
Inject rms-env after disable_komodo_exec is called

### PR DESCRIPTION
**Pull request by ManInFez**

### CODE REVIEW

**Comment by ManInFez** [comment]
Test this please!

File: res/fm/rms/rms_run.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def run(self):
         if os.path.isfile(exec_env_file):
             with open(exec_env_file) as f:
                 user_env = json.load(f)
+        print("USERENV", user_env)
```
Comment (sregales-TNO) [inline res/fm/rms/rms_run.py:115]: Do we really need these print statements?

Comment (sregales-TNO) [reply res/fm/rms/rms_run.py:115]: nevermind I see they are removed later on please drop this commit before rebasing

Comment (ManInFez) [reply res/fm/rms/rms_run.py:115]: nope ;)

Comment (sregales-TNO) [reply res/fm/rms/rms_run.py]: Other than that it looks good to me

File: res/fm/rms/rms_run.py [hunk 2]
```diff
@@ -164,7 +165,18 @@ def run(self):
...
     def exec_rms(self, exec_env):
-        args = [self.config.wrapper] if self.config.wrapper is not None else []
+        # The rms exec environement needs to be injected betweed executing the
+        # wrapper and launching rms. PATH_PREFIX must be set in advance.
+        prefix_path = exec_env.pop("PATH_PREFIX", "")
+        env_args = ["env"]
+        for key, value in exec_env.items():
+            env_args.append(f"{key}={value}")
+        args = (
+            ["env", f"PATH_PREFIX={prefix_path}", self.config.wrapper] + env_args
```
Comment (sregales-TNO) [inline res/fm/rms/rms_run.py:175]: would this line have a duplication and is it done on purpose `"env"`

File: res/fm/rms/rms_run.py [hunk 3]
```diff
@@ -164,7 +165,18 @@ def run(self):
             )
 
     def exec_rms(self, exec_env):
-        args = [self.config.wrapper] if self.config.wrapper is not None else []
+        # The rms exec environement needs to be injected betweed executing the
+        # wrapper and launching rms. PATH_PREFIX must be set in advance.
+        prefix_path = exec_env.pop("PATH_PREFIX", "")
+        env_args = ["env"]
+        for key, value in exec_env.items():
```
Comment (sregales-TNO) [inline res/fm/rms/rms_run.py:172]: suggestion (maybe less readable)
```python
env_args = ["env", **(f"{key}={value}" for key, value in )]
```

File: res/fm/rms/rms_run.py (hunk 2 above)
Comment (ManInFez) [reply res/fm/rms/rms_run.py:175]: so env should be called twice, once before the wrapper and once in between the wrapper and rms-executable

Comment (ManInFez) [reply res/fm/rms/rms_run.py:175]: The first one should just set the PATH_PREFIX

Comment (sregales-TNO) [reply res/fm/rms/rms_run.py:175]: understood

**Review by sregales-TNO — APPROVED** [review]

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by ManInFez** [comment]
Test this please!

**Comment by jondequinor** [comment]
This needs to be backported if it is to become a part of 2021.10.

**Pull request merged**